### PR TITLE
Add startTLS support

### PR DESCRIPTION
**Pull request by aperiodic**
Change the options to `create-connection` to look for a `tls?` key, in
addition to `ssl?`. If `tls?` is truthy, then have the connection
perform the startTLS invocation after it's created. If both `tls?` and
`ssl?` are truthy, throw an IllegalArgumentException, as the two are
mutually exclusive.

Add a test to ensure that we can connect to the server and perform the
startTLS invocation without errors (this is not a strong test that
startTLS works, but it's as strong as the SSL tests, and a truly strong
test would be much more work).

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -3,14 +3,13 @@
...
-  :dev-dependencies [[jline "0.9.94"]
-                     [org.apache.directory.server/apacheds-all "1.5.5"]
-                     [fs "1.1.2"]
-                     [org.slf4j/slf4j-simple "1.5.6"]
-                     [lein-clojars "0.7.0"]]
+  :profiles {:dev {:dependencies [[jline "0.9.94"]
+                                  [org.apache.directory.server/apacheds-all "1.5.5"]
+                                  [fs "1.1.2"]
+                                  [org.slf4j/slf4j-simple "1.5.6"]
+                                  [lein-clojars "0.7.0"]]}}
```
Comment (pcarlisle) [inline project.clj]: you can remove lein-clojars now too

File: src/clj_ldap/client.clj [hunk 2]
```diff
@@ -133,13 +147,25 @@
 
 (defn- create-connection
   "Create an LDAPConnection object"
-  [{:keys [host ssl?] :as options}]
+  [{:keys [host ssl? tls?] :as options}]
```
Comment (pcarlisle) [inline src/clj_ldap/client.clj]: `tls` is ambiguous, I think this should be `starttls`

**Comment by pcarlisle** [comment]
comment

**Comment by aperiodic** [comment]
@pcarlisle I removed lein-clojars and changed `tls?` to `start-tls?`

**Pull request merged**